DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/CentreConfigurationController.cs

```diff
@@ -80,13 +90,36 @@ public IActionResult EditCentreWebsiteDetails(EditCentreWebsiteDetailsViewModel
                 return View(model);
             }
 
+            model.CentrePostcode = model.CentrePostcode!.Trim();
+            var mapsResponse = mapsApiHelper.GetPostcodeCoordinates(model.CentrePostcode).Result;
+
+            if (mapsResponse.Status == "ZERO_RESULTS")
+            {
+                ModelState.AddModelError(nameof(model.CentrePostcode), "Enter a valid postcode.");
+                return View(model);
+            }
+
+            if (mapsResponse.Status != "OK")
+            {
+                logger.LogWarning
+                (
+                    $"Failed Maps API call when trying to get postcode {model.CentrePostcode} " +
+                    $"- status of {mapsResponse.Status} - error message: {mapsResponse.ErrorMessage}"
+                );
+                return RedirectToAction("Error", "LearningSolutions");
+            }
+
+            var longitude = double.Parse(mapsResponse.Results[0].Geometry.Location.Longitude);
+            var latitude = double.Parse(mapsResponse.Results[0].Geometry.Location.Latitude);
+
             var centreId = User.GetCentreId();
 
-            centresDataService.UpdateCentreWebsiteDetails
-            (
+            centresDataService.UpdateCentreWebsiteDetails(
                 centreId,
-                model.CentrePostcode!,
+                model.CentrePostcode,
                 model.ShowCentreOnMap,
+                longitude,
+                latitude,

```

Comment:
This is super minor, but my intuition is that latitude comes before longitude 😄

Reply:
Typically, yes. However, in the database longitude comes before latitude, so when writing this I kept everything in the same order for consistency sake. Do we want the code to be consistent with general standards and thus have latitude before longitude?

Reply:
I think we should just to avoid anyone accidentally writing lat, then long. The unexpected order (long, lat) is much easier to spot than that lat -> long order when reviewing for example :)